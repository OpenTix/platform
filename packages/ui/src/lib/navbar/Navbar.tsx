import { Container } from '@radix-ui/themes';
import React, { FC } from 'react';
import styled from 'styled-components';

const NavWrapper = styled.nav`
	width: 100%;
	background-color: #fff;
	border-bottom: 1px solid #ddd;
`;

const NavContent = styled.div`
	margin: 0 auto;
	padding: 0 16px;
	display: flex;
	align-items: center;
	height: 60px;
	justify-content: space-between;
`;

export interface NavbarProps {
	children: React.ReactNode;
}

export const Navbar: FC<NavbarProps> = ({ children }) => {
	return (
		<NavWrapper>
			<Container size={'4'}>
				<NavContent>{children}</NavContent>
			</Container>
		</NavWrapper>
	);
};
