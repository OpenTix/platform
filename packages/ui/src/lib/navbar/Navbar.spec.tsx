import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { Navbar } from './Navbar';

// Adjust path as needed

describe('Navbar', () => {
	it('renders without crashing', () => {
		render(<Navbar>Test Children</Navbar>);
		expect(screen.getByText('Test Children')).toBeInTheDocument();
	});

	it('renders children correctly', () => {
		render(
			<Navbar>
				<div data-testid="child-element">Child Content</div>
			</Navbar>
		);
		expect(screen.getByTestId('child-element')).toHaveTextContent(
			'Child Content'
		);
	});

	it('matches snapshot', () => {
		const { asFragment } = render(<Navbar>Snapshot Test</Navbar>);
		expect(asFragment()).toMatchSnapshot();
	});
});
