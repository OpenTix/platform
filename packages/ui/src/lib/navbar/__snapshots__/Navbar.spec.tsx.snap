// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="Navbar__NavWrapper-sc-1q1j1oe-0 fkYBon"
  >
    <div
      class="rt-Container rt-r-size-4"
    >
      <div
        class="rt-ContainerInner"
      >
        <div
          class="Navbar__NavContent-sc-1q1j1oe-1 jOdONA"
        >
          Snapshot Test
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
